[
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/data_connector.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/data_connector.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/data_connector.py:327:19 Incompatible parameter type [6]: Expected `typing.Sized` for 1st positional only parameter to call `len` but got `Optional[List[BatchDefinition]]`.",
    "message": " Expected `typing.Sized` for 1st positional only parameter to call `len` but got `Optional[List[BatchDefinition]]`.",
    "rule_id": "Incompatible parameter type [6]",
    "warning_line_no": 327,
    "warning_line": "        assert len(batch_definition_list) == 1"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:32:12 Incompatible parameter type [6]: Expected `typing.Callable[..., typing.Any]` for 1st parameter `custom_filter_function` to call `PartitionQuery.__init__` but got `None`.",
    "message": " Expected `typing.Callable[..., typing.Any]` for 1st parameter `custom_filter_function` to call `PartitionQuery.__init__` but got `None`.",
    "rule_id": "Incompatible parameter type [6]",
    "warning_line_no": 32,
    "warning_line": "            custom_filter_function=None,"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:35:12 Incompatible parameter type [6]: Expected `int` for 4th parameter `limit` to call `PartitionQuery.__init__` but got `None`.",
    "message": " Expected `int` for 4th parameter `limit` to call `PartitionQuery.__init__` but got `None`.",
    "rule_id": "Incompatible parameter type [6]",
    "warning_line_no": 35,
    "warning_line": "            limit=None,"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:44:4 Incompatible variable type [9]: custom_filter_function is declared to have type `typing.Callable[..., typing.Any]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "message": " custom_filter_function is declared to have type `typing.Callable[..., typing.Any]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 44,
    "warning_line": "    custom_filter_function: Callable = partition_request_dict.get("
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:53:4 Incompatible variable type [9]: batch_identifiers is declared to have type `Optional[Dict[typing.Any, typing.Any]]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "message": " batch_identifiers is declared to have type `Optional[Dict[typing.Any, typing.Any]]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 53,
    "warning_line": "    batch_identifiers: Optional[dict] = partition_request_dict.get(\"batch_identifiers\")"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:69:4 Incompatible variable type [9]: index is declared to have type `Union[None, typing.List[typing.Any], int, slice, str, typing.Tuple[typing.Any, ...]]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "message": " index is declared to have type `Union[None, typing.List[typing.Any], int, slice, str, typing.Tuple[typing.Any, ...]]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 69,
    "warning_line": "    index: Optional[Union[int, list, tuple, slice, str]] = partition_request_dict.get("
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:72:4 Incompatible variable type [9]: limit is declared to have type `Optional[int]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "message": " limit is declared to have type `Optional[int]` but is used as type `Union[typing.Callable[..., typing.Any], None, Dict[typing.Any, typing.Any], typing.List[typing.Any], PartitionDefinitionSubset, int, slice, str, typing.Tuple[typing.Any, ...]]`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 72,
    "warning_line": "    limit: Optional[int] = partition_request_dict.get(\"limit\")"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:87:8 Incompatible parameter type [6]: Expected `int` for 3rd parameter `limit` to call `PartitionQuery.__init__` but got `Optional[int]`.",
    "message": " Expected `int` for 3rd parameter `limit` to call `PartitionQuery.__init__` but got `Optional[int]`.",
    "rule_id": "Incompatible parameter type [6]",
    "warning_line_no": 87,
    "warning_line": "        limit=limit,"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:135:8 Incompatible variable type [9]: custom_filter_function is declared to have type `typing.Callable[..., typing.Any]` but is used as type `None`.",
    "message": " custom_filter_function is declared to have type `typing.Callable[..., typing.Any]` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 135,
    "warning_line": "        custom_filter_function: Callable = None,"
  },
  {
    "project": "great-expectations/great_expectations",
    "commit": "a4d6617d95b5b5317329df8637ab9a484dfe753a",
    "filename": "great_expectations/datasource/data_connector/partition_query.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/great-expectations-great_expectations/great_expectations/datasource/data_connector/batch_filter.py",
    "file_hunks_size": 19,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "great_expectations/datasource/data_connector/partition_query.py:138:8 Incompatible variable type [9]: limit is declared to have type `int` but is used as type `None`.",
    "message": " limit is declared to have type `int` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 138,
    "warning_line": "        limit: int = None,"
  }
]